### PR TITLE
SI-4829 the :load command now fails if the command ends with a space

### PR DESCRIPTION
**Pull request by jedesah**
The :load command now tries to be smart by detecting the presence of trailing spaces that might be insignificant.

On certain file systems, files with trailing whitespaces are permitted which is why we don't naievely trim the argument.

### CODE REVIEW

**Comment by som-snytt** [comment]
That's a clever approach.  I was going to try to get quotes working, because there's another ticket related to that, so that you would just quote it "my-evil-file- ".  I haven't looked it further yet, however.

What an interesting build error: No classes or objects found in package.scala that go in package scala

**Comment by jedesah** [comment]
Thx for the feedback!

While reviewing the code to make the ajustements you suggested, I realized that the addClasspath method had very similar code and that there might be a way to better factor things. I used the error message from the addClasspath, but of course we can change it if we think another formulation might be better.

**Comment by jedesah** [comment]
I am not sure I am understanding these build errors. Does the build machine handle the 2.9.x branch?

I am unable to reproduce the build error on my machine even with a clean rebuild.

Any ideas?

**Comment by jedesah** [comment]
@som-snytt What ticket is that? I wouldn't mind having a look at it.

**Comment by som-snytt** [comment]
@jedesah The ticket about quotes was https://issues.scala-lang.org/browse/SI-6659 which I just assigned to myself, it's actually tab completion.  I don't have code yet.  It would nice to have tab completion on file commands, too, load fo\t.

File: src/compiler/scala/tools/util/StringOps.scala [hunk 1]
```diff
@@ -54,6 +54,10 @@ trait StringOps {
     if (idx == -1) None
     else Some(str take idx, str drop (if (doDropIndex) idx + 1 else idx))
 
+  def splitLeft(str: String, ch: Char): Option[(String, String)] = {
```
Comment (dragos) [inline src/compiler/scala/tools/util/StringOps.scala:57]: `ch` is not used.

**Comment by dragos** [comment]
The range-pos build errors can be ignored, they always fail on 2.9.

File: src/compiler/scala/tools/util/StringOps.scala (hunk 1 above)
Comment (jedesah) [inline src/compiler/scala/tools/util/StringOps.scala:57]: Thank you for the catch! :-)

**Comment by adriaanm** [comment]
LGTM -- would you like to port this to 2.10.x/master as well?

**Comment by jedesah** [comment]
@adriaanm Sure, I can do that.

**Pull request merged**